tuta-sdk/rust/sdk/src/entities/entity_facade.rs

```diff
@@ -232,66 +210,80 @@ impl EntityFacadeImpl {
-					.get_type_model(dependency, association.ref_type)
-					.unwrap_or_else(|| {
-						panic!(
-							"Cannot find type model for: {:?}",
-							(dependency, association.ref_type)
-						)
-					});
-				let aggregation = association;
-				let instance_association = instance.get(&association_name.to_string()).unwrap();
-				if aggregation.cardinality == Cardinality::ZeroOrOne
-					&& instance_association == &ElementValue::Null
-				{
-					encrypted.insert(association_name.to_string(), ElementValue::Null);
-				} else if instance_association == &ElementValue::Null {
-					panic!("Undefined attribute {}:{association_name}", type_model.name);
-				} else if aggregation.cardinality == Cardinality::Any {
-					let aggregates = instance_association.assert_array();
-					let mut encrypted_aggregates = Vec::with_capacity(aggregates.len());
-					for aggregate in &aggregates {
-						let parsed_entity = self.encrypt_and_map_to_literal_with_iv(
-							aggregated_type_model,
-							&aggregate.assert_dict(),
-							sk.clone(),
-							random_iv_provider.clone(),
-						)?;
-						encrypted_aggregates.push(ElementValue::Dict(parsed_entity));
-					}
+			let encrypted_association = match association.association_type {
+				AssociationType::Aggregation => self.encrypt_aggregate(
+					type_model,
+					association_name,
+					association,
+					instance,
+					sk.as_ref(),
+				)?,
+				AssociationType::ElementAssociation
+				| AssociationType::ListAssociation
+				| AssociationType::ListElementAssociation
+				| AssociationType::BlobElementAssociation => instance
+					.get(&association_name.to_string())
+					.cloned()
+					.ok_or(ApiCallError::internal(format!(
+						"could not find association {association_name} on type {}",
+						type_model.name
+					)))?,
+			};
+			encrypted.insert(association_name.to_string(), encrypted_association);
+		}
 
-					encrypted.insert(
-						association_name.to_string(),
-						ElementValue::Array(encrypted_aggregates),
-					);
-				} else {
-					let parsed_entity = self.encrypt_and_map_to_literal_with_iv(
+		Ok(encrypted)
+	}
+
+	fn encrypt_aggregate(
+		&self,
+		type_model: &TypeModel,
+		association_name: &str,
+		association: &ModelAssociation,
+		instance: &ParsedEntity,
+		sk: Option<&GenericAesKey>,
+	) -> Result<ElementValue, ApiCallError> {
+		let dependency = association.dependency.unwrap_or(type_model.app);
+		let aggregated_type_model = self
+			.type_model_provider
+			.get_type_model(dependency, association.ref_type)
+			.unwrap_or_else(|| {
+				panic!(
+					"Cannot find type model for: {:?}",
+					(dependency, association.ref_type)
+				)
+			});
+		let instance_association = instance.get(&association_name.to_string()).unwrap();
+
+		match (&association.cardinality, instance_association) {
+			(Cardinality::ZeroOrOne, ElementValue::Null) => Ok(ElementValue::Null),
+			(_, ElementValue::Null) => {
+				panic!("Undefined attribute {}:{association_name}", type_model.name)
```

Comment:
when we can't find association it is `Err` but here we `panic()`. We should be consistent.